### PR TITLE
Fix some group-related code to handle CamelCase usernames

### PR DESCRIPTION
**Pull request by sethalves**
- Fix some group-related code to handle CamelCase usernames

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- put someone with some capital letters in their name who cannot otherwise connect into a group that can connect
- make sure they can now connect

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3ca1ee1d82ca9292ad2e26a5d907c3e576332887 available at: https://gist.github.com/36ed942ec4ec2d1578ba13f3020638ee<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8373-3ca1ee1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8373-3ca1ee1.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7aa7cdf64225dff5404a514063fee9d8fd427a08 available at: https://gist.github.com/36ed942ec4ec2d1578ba13f3020638ee<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8373-7aa7cdf.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8373-7aa7cdf.dmg

**Comment by cozza13** [comment]
This was tested by Thoys and Piper.

File: domain-server/src/DomainServerSettingsManager.cpp [hunk 1]
```diff
@@ -356,7 +356,7 @@ void DomainServerSettingsManager::initializeGroupPermissions(NodePermissionsMap&
         if (nameKey.first.toLower() != groupNameLower) {
             continue;
         }
-        QUuid groupID = _groupIDs[groupNameLower];
+        QUuid groupID = _groupIDs[groupNameLower.toLower()];
```
Comment (huffman) [inline domain-server/src/DomainServerSettingsManager.cpp]: @sethalves Actually, noticed this after, this toLower() is redundant

Comment (sethalves) [inline domain-server/src/DomainServerSettingsManager.cpp]: you can never be too sure

**Pull request merged**